docs/user-guide/config-best-practices.md

```diff
@@ -31,27 +31,131 @@ Documentation for other releases can be found at
-1. Put an object description in an annotation to allow better introspection.
+# Configuration Best Practices and Tips
+
+**Table of Contents**
+<!-- BEGIN MUNGE: GENERATED_TOC -->
+
+- [Configuration Best Practices and Tips](#configuration-best-practices-and-tips)
+  - [General Config Tips](#general-config-tips)
+  - ["Naked" Pods vs Replication Controllers and Jobs](#naked-pods-vs-replication-controllers-and-jobs)
+  - [Services](#services)
+  - [Using Labels](#using-labels)
+  - [Container Images](#container-images)
+  - [Using kubectl](#using-kubectl)
+
+<!-- END MUNGE: GENERATED_TOC -->
+
+
+This document is meant to highlight and consolidate in one place configuration best practices that are introduced throughout the user-guide and getting-started documentation and examples. This is a living document, so if you think of something that is not on this list but might be useful to others, please don't hesitate to file an issue or submit a PR.
+
+## General Config Tips
+
+- When defining configurations, specify the latest stable API version (currently v1).
+
+- Configuration files should be stored in version control before being pushed to the cluster. This allows a configuration to be quickly rolled back if needed, and will aid with cluster re-creation and restoration if necessary.
+
+- Write your configuration files using YAML rather than JSON. They can be used interchangeably in almost all scenarios, but YAML tends to be more user-friendly for config.
+
+- Group related objects together in a single file where this makes sense. This format is often easier to manage than separate files. See the [guestbook-all-in-one.yaml](../../examples/guestbook/all-in-one/guestbook-all-in-one.yaml) file as an example of this syntax.
+(Note that you can also call `kubectl create` on a directory of config files— see below for more detail).
+
+- Don't specify default values unnecessarily, in order to simplify and minimize configs, and to
+  reduce error. For example, omit the selector and labels in a `ReplicationController` if you want
+  them to be the same as the labels in its `podTemplate`, since those fields are populated from the
+  `podTemplate` labels by default. See the [guestbook app's](../../examples/guestbook/) .yaml files for some [examples](../../examples/guestbook/frontend-controller.yaml) of this.
+
+- Put an object description in an annotation to allow better introspection.
+
+
+## "Naked" Pods vs Replication Controllers and Jobs
+
+- If there is a viable alternative to naked pods (i.e., pods not bound to a controller), go with
+  the alternative. Naked pods will not be rescheduled in the event of node failure.
+
+  Controllers are almost always preferable to creating pods, except for some explicit
+  [`restartPolicy: Never`](pod-states.md#restartpolicy) scenarios.  A
+  [Job](jobs.md) object (currently in Beta), may also be appropriate. A _job_ creates one or more
```

Comment:
This part contains too much jobs and too few replication controllers information. Just present enough information for why we should avoid naked pods and link to both documents for more details.